desc/compute/_equil.py

```diff
@@ -4,118 +4,423 @@
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Contravariant toroidal component of plasma current density",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["sqrt(g)", "B_theta_r", "B_rho_t"],
+)
+def _J_sup_zeta(params, transforms, profiles, data, **kwargs):
+    data["J^zeta"] = (data["B_theta_r"] - data["B_rho_t"]) / (mu_0 * data["sqrt(g)"])
+    return data
+
+
+@register_compute_fun(
+    name="J",
+    label="\\mathbf{J}",
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Plasma current density",
+    dim=3,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["J^rho", "J^theta", "J^zeta", "e_rho", "e_theta", "e_zeta"],
+)
+def _J(params, transforms, profiles, data, **kwargs):
+    data["J"] = (
+        data["J^rho"] * data["e_rho"].T
+        + data["J^theta"] * data["e_theta"].T
+        + data["J^zeta"] * data["e_zeta"].T
+    ).T
+    return data
+
+
+@register_compute_fun(
+    name="J_R",
+    label="J_{R}",
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Radial componenet of plasma current density in lab frame",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["J"],
+)
+def _J_R(params, transforms, profiles, data, **kwargs):
+    data["J_R"] = data["J"][:, 0]
+    return data
+
+
+@register_compute_fun(
+    name="J_phi",
+    label="J_{\\phi}",
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Toroidal componenet of plasma current density in lab frame",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["J"],
+)
+def _J_phi(params, transforms, profiles, data, **kwargs):
+    data["J_phi"] = data["J"][:, 1]
+    return data
+
+
+@register_compute_fun(
+    name="J_Z",
+    label="J_{Z}",
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Vertical componenet of plasma current density in lab frame",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["J"],
+)
+def _J_Z(params, transforms, profiles, data, **kwargs):
+    data["J_Z"] = data["J"][:, 2]
+    return data
+
+
+@register_compute_fun(
+    name="|J|",
+    label="|\\mathbf{J}|",
+    units="A \\cdot m^{-3}",
+    units_long="Amperes / cubic meter",
+    description="Magnitue of plasma current density",
+    dim=1,
+    params=[],
+    transforms={},
+    profiles=[],
+    function_of="rtz",
+    data=["J"],
+)
+def _Jmag(params, transforms, profiles, data, **kwargs):
+    data["|J|"] = jnp.linalg.norm(data["J"], axis=-1)
+    return data
+
+
+@register_compute_fun(
+    name="J*B",
+    label="\\mathbf{J} \\cdot \\mathbf{B}",
+    units="N / m^{3}",
+    units_long="Newtons / cubic meter",
+    description="Bootstrap current",
```

Comment:
Maybe worth mentioning this is NOT in units of current density in the description?